velox/exec/tests/HashJoinTest.cpp

```diff
@@ -46,43 +71,80 @@ class HashJoinTest : public HiveConnectorTestBase {
   void testJoin(
       const std::vector<TypePtr>& keyTypes,
       int32_t numThreads,
-      int32_t leftSize,
-      int32_t rightSize,
+      int32_t leftBatchSize,
+      int32_t rightBatchSize,
       const std::string& referenceQuery,
       const std::string& filter = "") {
     auto leftType = makeRowType(keyTypes, "t_");
     auto rightType = makeRowType(keyTypes, "u_");
 
-    auto leftBatch = std::dynamic_pointer_cast<RowVector>(
-        BatchMaker::createBatch(leftType, leftSize, *pool_));
-    auto rightBatch = std::dynamic_pointer_cast<RowVector>(
-        BatchMaker::createBatch(rightType, rightSize, *pool_));
-
-    CursorParameters params;
-    auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
-    params.planNode = PlanBuilder(planNodeIdGenerator)
-                          .values({leftBatch}, true)
-                          .hashJoin(
-                              makeKeyNames(keyTypes.size(), "t_"),
-                              makeKeyNames(keyTypes.size(), "u_"),
-                              PlanBuilder(planNodeIdGenerator)
-                                  .values({rightBatch}, true)
-                                  .planNode(),
-                              filter,
-                              concat(leftType->names(), rightType->names()))
-                          .planNode();
-    params.maxDrivers = numThreads;
-
-    createDuckDbTable("t", {leftBatch});
-    createDuckDbTable("u", {rightBatch});
-    auto task = ::assertQuery(
-        params, [](auto*) {}, referenceQuery, duckDbQueryRunner_);
-
-    // A quick sanity check for memory usage reporting. Check that peak total
-    // memory usage for the hash join node is > 0.
-    auto planStats = toPlanStats(task->taskStats());
-    auto joinNodeId = params.planNode->id();
-    ASSERT_TRUE(planStats.at(joinNodeId).peakMemoryBytes > 0);
+    std::vector<RowVectorPtr> leftBatches;
+    std::vector<RowVectorPtr> rightBatches;
+    constexpr int numLeftBatches = 4;
```

Comment:
Do we want to also make numLeftBatches and numRightBatches configurable for more coverage or we are good this way? If we set them to fixed numbers does it make sense we declare them on top of the class?